onnxscript/version_converter/version_converter.py

```diff
@@ -0,0 +1,310 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+"""Convert the model to the specified ONNX opset version."""
+
+from __future__ import annotations
+
+import dataclasses
+import logging
+from typing import Callable, Sequence, Union
+
+import onnxscript.ir._convenience as _convenience
+import onnxscript.rewriter.pattern as orp
+from onnxscript import ir
+
+logger = logging.getLogger(__name__)
+
+
+CURRENT_MAX_ONNX_OPSET = 23
+
+
+@dataclasses.dataclass
+class Replacement:
+    """A replacement for a node in the graph."""
+
+    new_outputs: Sequence[ir.Value]
+    new_nodes: Sequence[ir.Node]
+
+
+# A version-adapter function takes a node, a RewriterContext and returns
+# a Replacement for the node or None (if no replacement is needed).
+
+ReturnValue = Union[Replacement, Sequence[ir.Value], ir.Value, None]
```

Comment:
```suggestion
_ReturnValue = Union[Replacement, Sequence[ir.Value], ir.Value, None]
```

make private. Everything not in `__all__` should be private